src/evict/evict_page.c

```diff
@@ -266,20 +283,25 @@ __wt_evict(WT_SESSION_IMPL *session, WT_REF *ref, uint8_t previous_state, uint32
 
         WT_STAT_CONN_DATA_INCR(session, cache_eviction_fail);
     }
+    if (!session->evict_timeline.reentry_hs_eviction) {
+        eviction_time_milliseconds = eviction_time / WT_THOUSAND;
+        if (eviction_time_milliseconds > conn->cache->evict_max_ms)
+            conn->cache->evict_max_ms = eviction_time_milliseconds;
+        if (eviction_time_milliseconds > 60000)
+            __wt_verbose_warning(session, WT_VERB_EVICT,
+              "Eviction took more than 1 minute (%" PRIu64 "). Building disk image took %" PRIu64
+              "us. History store wrapup took %" PRIu64 "us.",
+              eviction_time,
+              WT_CLOCKDIFF_US(session->reconcile_timeline.image_build_finish,
+                session->reconcile_timeline.image_build_start),
+              WT_CLOCKDIFF_US(session->reconcile_timeline.hs_wrapup_finish,
+                session->reconcile_timeline.hs_wrapup_start));
+    } else {
+        session->reconcile_timeline.total_nested_eviction_time += WT_CLOCKDIFF_MS(
```

Comment:
This deserves a comment please - describing why nested is special.

Reply:
done